crates/astria-composer/src/searcher/collector.rs

```diff
@@ -0,0 +1,221 @@
+use std::time::Duration;
+
+use color_eyre::eyre::{
+    self,
+    WrapErr as _,
+};
+use ethers::providers::{
+    Provider,
+    ProviderError,
+    Ws,
+};
+use humantime::format_duration;
+use tokio::sync::{
+    mpsc::{
+        error::SendTimeoutError,
+        Sender,
+    },
+    watch,
+};
+use tracing::{
+    debug,
+    info,
+    instrument,
+    warn,
+};
+
+/// A wrapper around an [`ethers::types::Transaction`] that includes the chain ID.
+///
+/// Used to send new transactions to the searcher.
+pub(super) struct Transaction {
```

Comment:
Nit: Can we add a `new` function to this to build descriptively?

Reply:
I don't see the need for that. This type is constructed only in one place.